indigo_social/models.py

```diff
@@ -25,3 +54,17 @@ def create_user_profile(sender, **kwargs):
     if not hasattr(user, "userprofile") and not kwargs.get("raw"):
         user_profile = UserProfile(user=user)
         user_profile.save()
+
+
+@receiver(user_signed_up, sender=User)
+def retrieve_profile_photo_on_signup(sender, **kwargs):
+    user = kwargs['user']
+    user_profile = UserProfile.objects.get(user=user)
```

Comment:
This will throw `UserProfile.DoesNotExist` if the object doesn't exist. Either handle the exception, or use `.filter(...).first()`.